pkg/irreparable/irreparable_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package irreparable
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+	"storj.io/storj/internal/testcontext"
+)
+
+func TestIrreparableRepository(t *testing.T) {
+	//testing variables
+	var (
+		rmtsegpath         = []byte("IamSegmentkeyinfo")
+		rmtsegdetail       = []byte("IamSegmentdetailinfo")
+		piecesLost         = int64(10)
+		damagedsegUnixsec  = time.Now().Unix()
+		repairAttemptCount = int64(10)
+	)
+
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	// creating in-memory db and opening connection
+	irrdb, err := newTestIrreparableDB()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ctx.Check(irrdb.DB.Close)
+
+	t.Run("New Entry", func(t *testing.T) {
+		rmtseginfo := &RemoteSegmentInfo{
+			SegmentPath:        rmtsegpath,
+			SegmentDetail:      rmtsegdetail,
+			LostPiecesCount:    piecesLost,
+			RepairUnixSec:      damagedsegUnixsec,
+			RepairAttemptCount: repairAttemptCount,
+		}
+
+		err = irrdb.Put(ctx, rmtseginfo)
+		assert.NoError(t, err)
+	})
+
+	t.Run("Entry already exists", func(t *testing.T) {
+		createReq := &RemoteSegmentInfo{
+			SegmentPath:        rmtsegpath,
+			SegmentDetail:      rmtsegdetail,
+			LostPiecesCount:    piecesLost,
+			RepairUnixSec:      damagedsegUnixsec,
+			RepairAttemptCount: repairAttemptCount,
+		}
+
+		err = irrdb.Put(ctx, createReq)
+		assert.NoError(t, err)
+
+		dbrmtsegInfo, err := irrdb.Get(ctx, createReq)
```

Comment:
`dbrmtsegInfo` <- I'm not sure what this is

Reply:
This is the database row returned, based on 'Segmentpath' information, which is passed in the argument , if there is an entry with that path already or else it returns no rows found, basically err != nil. Renamed it to be more intuitive... renamed it as 'segmentInfoRow'.